src/main/java/org/currencygoldexchangeapp/services/UserMenuService.java

```diff
@@ -0,0 +1,177 @@
+import org.currencygoldexchangeapp.datamodels.GoldValue;
+import org.currencygoldexchangeapp.exceptions.CurrencyNotFoundException;
+import org.currencygoldexchangeapp.exceptions.DataNotFoundException;
+import org.currencygoldexchangeapp.handlers.ExchangeRateAPIHandler;
+import org.currencygoldexchangeapp.handlers.ExchangeRateFileReaderHandler;
+import org.currencygoldexchangeapp.handlers.ExchangeRateFileSaverHandler;
+import org.currencygoldexchangeapp.handlers.GoldValueAPIHandler;
+import org.currencygoldexchangeapp.utils.InputUtility;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.http.HttpClient;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Scanner;
+
+public class UserMenuService {
+    private final Scanner scanner = new Scanner(System.in);
+    private final ExchangeRateFileReaderHandler exchangeRateFileReaderHandler;
+    private final ExchangeRateFileSaverHandler exchangeRateFileSaverHandler;
+    private final GoldValueAPIHandler goldValueAPIHandler;
+    private final CurrencyExchangeCalculateService currencyExchangeCalculateService;
+    private final GoldValueCalculateService goldValueCalculateService;
+
+    public UserMenuService() {
+        ExchangeRateAPIHandler exchangeRateAPIHandler = new ExchangeRateAPIHandler(HttpClient.newHttpClient());
+        this.currencyExchangeCalculateService = new CurrencyExchangeCalculateService(exchangeRateAPIHandler);
+        this.exchangeRateFileReaderHandler = new ExchangeRateFileReaderHandler(currencyExchangeCalculateService);
+        this.exchangeRateFileSaverHandler = new ExchangeRateFileSaverHandler();
+        this.goldValueAPIHandler = new GoldValueAPIHandler(HttpClient.newHttpClient());
+        this.goldValueCalculateService = new GoldValueCalculateService(goldValueAPIHandler);
+    }
+
+    public void runApplication() {
+        int choice = 0;
+        while (choice != 5) {
+            System.out.println();
+            displayMenu();
+            choice = getUserChoice();
+            System.out.println();
+            handleUserChoice(choice);
+        }
+    }
+
+    private void displayMenu() {
+        System.out.println("1. Currency exchange prices for a specific date.");
+        System.out.println("2. Load data from a file and calculate currency exchange prices.");
+        System.out.println("3. Gold price for a specific date.");
+        System.out.println("4. Compare today's gold price with the best price from the last month/year.");
+        System.out.println("5. Close the application");
+    }
+
+    private int getUserChoice() {
+        System.out.print("Please enter your choice: ");
+        String input = scanner.nextLine();
+        try {
+            return Integer.parseInt(input);
+        } catch (NumberFormatException e) {
+            System.out.println("Conversion error: " + e.getMessage());
```

Comment:
It would be better to just let user know, what needs to be done to continue. User does not care about conversion errors